content/reference/server-rendering.md

```diff
@@ -0,0 +1,124 @@
+[choojs/bankai](https://github.com/choojs/bankai). We recommend using Bankai if
+you're looking for a solution to Server Rendering. However, even if you end up
+using a prebuilt solution, it can still be useful to know how the underlying
+mechanisms work._
+
+## Rendering Choo in Node
+The main method to use in Choo for server rendering is `.toString(routeName)`.
+Using it looks something like this:
+
+```js
+var html = require('choo/html')
+var choo = require('choo')
+
+var app = choo()                    // 1.
+app.route('/', (state, emit) => {   // 2.
+  return html`
+    <body>Hello humans</body>
+  `
+})
+
+var dom = app.toString('/')         // 3.
+console.log(dom)
+```
+
+1. Create a new Choo app instance.
+2. Create a new route, it returns some HTML.
+3. Here is where the rendering happens. We call our application with
+   `.toString(route)`, and it returns a string. The resulting string will be
+   `'<body>Hello humans</body>'`.
+
+## Adding in Data
+There hasn't been too much going on so far. Instead of rendering to the DOM,
+we've been creating strings instead. However, what's often the case with server
+rendering is that you need some data passed in for your initial rendering.
+Luckily the `.toString()` method accepts a seonc argument: `state`. Let's see
+how this works.
+
+```js
+var html = require('choo/html')
+var choo = require('choo')
+
+var app = choo()
+var state = { message: 'doggos' }           // 1.
+app.route('/', (state, emit) => {         // 2.
+  return html`
+    <body>Hello ${state.message}</body>
+  `
+})
+
+var dom = app.toString('/', state)        // 3.
+console.log(dom)
+```
+
+1. We create a `state` object with some values in it.
+2. We render a `<body>` element. It takes a value from the `state`.
+3. We pass the `state` object to `app.toString()` as the second argument. It
+   returns a string of `<body>Hello doggos</body>`.
+
+## Isomorphic Rendering.
+In JavaScript the term "isomorphic code" often refers to code that can run in
+both Node and the Browser. For Choo code to be isomorphic, it should render in
+the browser, and should export the `app` instance in Node so it can be called
+for server rendering.
+
+### Classic
+The "classic" version assumes we use `require()` for our application.
+```js
+var choo = require('choo')
+
+var app = choo()
+app.route('/', () => html`<body>hello world</body>`)
+
+if (!module.parent) {   // 1.
+  app.mount('body')
+} else {                // 2.
+  module.exports = app
+}
+```
+
+1. If the current file isn't required by any other file, we `mount` the app on
+ the `<body>` tag.
+2. If the current file is required by some other file, we export it instead.
+
+### Modern
+The `import` keyword is what's going to be used to load JavaScript modules in
+the near future. It already works in Node, but isn't quite prevalent yet.
+Because the details of `import` are different from `require`, using it requires
+a slightly different approach.
+
+```js
+var choo = require('choo')
```

Comment:
isn't it supposed to use `import` here then?

Reply:
lol yup